### PR TITLE
fix: making sure writing to disk has finished before reading

### PR DESCRIPTION
**Pull request by firecode9**
- [X ] Ready for review
- [ X] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [ ] Reviewed by Snyk internal team

#### What does this PR do?
The wizard is trying to read from the disk before we are making sure that writing has finished. This will cause the process to fail later on

### CODE REVIEW

File: src/cli/commands/protect/wizard.ts [hunk 1]
```diff
@@ -524,7 +524,7 @@ function processAnswers(answers, policy, options) {
           options.packageTrailing;
         return (
           spinner(lbl)
-            .then(fs.writeFile(packageFile, packageString))
+            .then(fs.writeFileSync(packageFile, packageString))
```
Comment (lili2311) [inline src/cli/commands/protect/wizard.ts]: the `.then` should be waiting on the `fs.writeFile` to complete. What is the bug? Where is a test to show the problem please?

Comment (lili2311) [reply src/cli/commands/protect/wizard.ts]: please check the tests they are failing

**Comment by firecode9** [comment]
Related support ticket:
https://snyk.zendesk.com/agent/tickets/4288

File: src/cli/commands/protect/wizard.ts (hunk 1 above)
Comment (firecode9) [reply src/cli/commands/protect/wizard.ts]: There is a race condition here:
image

The write to disk promise is done however the content itself is still not written to disk. Later on in: `/src/lib/plugins/nodejs-plugin/npm-lock-parser.ts`

image

We are trying to read before we have the file on the disk

File: src/cli/commands/protect/wizard.ts [hunk 2]
```diff
@@ -68,16 +69,18 @@ async function processPackageManager(options: Options) {
...
-    if (!nodeModulesExist) {
-      // throw a custom error
-      throw new Error(
-        "Missing node_modules folder: we can't patch without having installed packages." +
-          `\nPlease run '${packageManager} install' first.`,
-      );
-    }
-    return options;
-  });
+  return fsPromise
```
Comment (dkontorovskyy) [inline src/cli/commands/protect/wizard.ts]: Can you make `async/await` this function?

File: src/cli/commands/protect/wizard.ts [hunk 3]
```diff
@@ -524,7 +527,10 @@ function processAnswers(answers, policy, options) {
           options.packageTrailing;
         return (
           spinner(lbl)
-            .then(fs.writeFile(packageFile, packageString))
+            .then(() => {
```
Comment (dkontorovskyy) [inline src/cli/commands/protect/wizard.ts]: Can you explain how this fixes the issue? Missing a bit.

Comment (firecode9) [reply src/cli/commands/protect/wizard.ts]: See here:
https://github.com/snyk/snyk/pull/1049#discussion_r395754744

**Pull request merged**